### PR TITLE
Add xxhash64 support

### PR DESCRIPTION
**Pull request by nvdbaranec**
Adds support for xxhash64.  Includes JNI bindings, cpp tests and java tests.  

Does not currently support structs or lists, but those should be relatively easy to add if desired.

<s>Dependent on https://github.com/NVIDIA/spark-rapids-jni/pull/1246</s>

### CODE REVIEW

File: src/test/java/com/nvidia/spark/rapids/jni/HashTest.java
Comment (jlowe) [inline src/test/java/com/nvidia/spark/rapids/jni/HashTest.java:339]:
```suggestion

  @Test
  void testXXHash64Decimal32() {
    // The hash values were derived from Apache Spark in a manner similar to the one documented at
    // https://github.com/rapidsai/cudf/blob/aa7ca46dcd9e/cpp/tests/hashing/hash_test.cpp#L281-L307
    try (ColumnVector v = ColumnVector.decimalFromInts(-3,
        0, 100, -100, 0x12345678, -0x12345678);
         ColumnVector result = Hash.xxhash64(new ColumnVector[]{v});
         ColumnVector expected = ColumnVector.fromBoxedLongs(-5252525462095825812, 8713583529807266080, 5675770457807661948, -7728554078125612835, 3142315292375031143)) {
      assertColumnsAreEqual(expected, result);
    }
  }
```

**Review by jlowe — COMMENTED** [review]
Marking as draft since this depends on #1244 and #1246.  This also needs a signoff commit per the contribution guidelines.

File: src/main/java/com/nvidia/spark/rapids/jni/Hash.java
Comment (jlowe) [inline src/main/java/com/nvidia/spark/rapids/jni/Hash.java:19]: Nit: This shouldn't include _everything_ from cudf.  There's very little that needs to be pulled into cudf here.

File: src/main/cpp/src/HashJni.cpp
Comment (jlowe) [inline src/main/cpp/src/HashJni.cpp:31]:
```suggestion
    cudf::jni::auto_set_dwevice(env);
    auto column_views =
```

File: src/main/cpp/src/HashJni.cpp
Comment (jlowe) [inline src/main/cpp/src/HashJni.cpp:47]:
```suggestion
    cudf::jni::auto_set_device(env);
    auto column_views =
```

File: src/main/cpp/src/HashJni.cpp
Comment (jlowe) [inline src/main/cpp/src/HashJni.cpp:26]:
```suggestion
  JNIEnv* env, jclass, jint seed, jlongArray column_handles)
```

File: src/main/cpp/src/HashJni.cpp
Comment (jlowe) [inline src/main/cpp/src/HashJni.cpp:40]:
```suggestion
                                                                       jclass,
```

**Review by jlowe — COMMENTED** [review]
lgtm, just need to get the dependencies merged first.

File: pom.xml
Comment (jlowe) [inline pom.xml:198]: Curious why pom.xml is being changed here?  This is unrelated to xxhash64 changes and effectively reverts #1252.

**Review by jlowe — CHANGES_REQUESTED** [review]

File: thirdparty/cudf
Comment (nvdbaranec) [reply thirdparty/cudf:1]: I'm not sure what happened here.  Probably an issue stemming from merging in 23.08 after the murmur hash move PR got merged.  I'll fix.

**Review by jlowe — APPROVED** [review]

File: src/main/cpp/src/xxhash64.cu
Comment (hyperbolic2346) [inline src/main/cpp/src/xxhash64.cu:76]:
```suggestion
                                                        cudf::size_type const nbytes,
```

File: src/test/java/com/nvidia/spark/rapids/jni/HashTest.java
Comment (hyperbolic2346) [inline src/test/java/com/nvidia/spark/rapids/jni/HashTest.java:274]: Is there a reason we don't use the CPU to hash the same ColumnVector and compare?

**Review by hyperbolic2346 — CHANGES_REQUESTED** [review]
Good work on this. I appreciate the tests added for murmur as well. A few nits and questions.

File: src/test/java/com/nvidia/spark/rapids/jni/HashTest.java
Comment (nvdbaranec) [reply src/test/java/com/nvidia/spark/rapids/jni/HashTest.java:274]: Not particularly.  Just following what the murmur hash tests do.

**Review by hyperbolic2346 — APPROVED** [review]

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

File: src/main/cpp/src/xxhash64.cu
Comment (rwlee) [inline src/main/cpp/src/xxhash64.cu:55]: Is there a functional difference that I'm missing between this and https://github.com/NVIDIA/spark-rapids-jni/pull/1246/files#diff-3b5fbfa28ce0798e562307d27b7636b7fd46e6d90d92e93c61a0da5411aec0dbR55. The addition of the extra static I believe is unnecessary.

**Pull request merged**
